src/test/resources/regressions/issues/000511.gobra

```diff
@@ -0,0 +1,34 @@
+// Any copyright is dedicated to the Public Domain.
+// http://creativecommons.org/publicdomain/zero/1.0/
+
+package issues
+
+preserves acc(x)
+ensures   *x == old(*x) + 1
+ensures !res
+func incF(x *int) (res bool) {
+	*x += 1
+	return false
+}
+
+func clientAdd() {
+	x@ := 1
+	if incF(&x) && incT(&x) {
+		assert false
+	}
+	assert x == 2
+}
+
+preserves acc(x)
+ensures   *x == old(*x) + 1
+ensures res
+func incT(x *int) (res bool) {
+	*x += 1
+	return true
+}
+
+func clientOr() {
+	x@ := 1
+	if incT(&x) || incT(&x) {}
```

Comment:
Please, also add cases where both calls happen, thus the variable is increased twice